src/logic/Application.swift

```diff
@@ -125,9 +125,12 @@ class Application: NSObject {
         }
     }
 
-    func quit() {
+    func quit(_ alertSound: Bool = false) {
         // only let power users quit Finder if they opt-in
-        if runningApplication.bundleIdentifier == "com.apple.finder" && !Preferences.finderShowsQuitMenuItem { return }
+        if runningApplication.bundleIdentifier == "com.apple.finder" && !Preferences.finderShowsQuitMenuItem {
+            if alertSound { NSSound.beep() }
```

Comment:
I'm not sure it's a good idea to move these checks upstream, and this solution seems more practical than duplicating them.

Reply:
I tested, and there is another issue upstream: some shortcuts don't trigger the sound naturally. Modifiers-only shortcuts for instance don't trigger it. It's a similar situation with the key-repeat, where we create a `NSTimer` instead on relying on the natural key repeat because some modifiers don't repeat naturally, so we create it artificially.

So in the end I think you approach here is the best 👍